enhanced-resolve/lib/Storage.d.ts

```diff
@@ -0,0 +1,30 @@
+/// <reference types="node" />
+import { CommonFileSystemMethod } from './common-types';
+
+declare class Storage {
+    duration: number;
+    running: Object;
+    data: Object;
+    levels: string[][];
+    count: number;
+    interval: NodeJS.Timer | null;
+    needTickCheck: boolean;
+    nextTick: number | null;
+    passive: boolean;
+
+    constructor(duration: number);
+
+    ensureTick(): void;
+
+    finished(name: string): void;
+
+    provide(name: string, provider: CommonFileSystemMethod, callback: (...args: any[]) => any): any;
```

Comment:
Even in this situation, I would avoid using `Function`, because `Function` is frequently used as a placeholder for a better function type. If you know that no better type can be used, then `(...args: any[]) => any` is fine.